pkg/gadgets/trace/network/tracer/tracer.go

```diff
@@ -256,21 +249,25 @@ func (t *Tracer) Pop() ([]*types.Event, error) {
 	return events, nil
 }
 
-func (t *Tracer) releaseLink(key string, l *link) {
+func (t *Tracer) releaseLink(netns uint64, l *link) {
 	unix.Close(l.sockFd)
 	l.networkGraphObjects.Close()
-	delete(t.attachments, key)
+	delete(t.attachments, netns)
 }
 
-func (t *Tracer) Detach(key string) error {
-	if l, ok := t.attachments[key]; ok {
+func (t *Tracer) Detach(pid uint32) error {
+	netns, err := containerutils.GetNetNs(int(pid))
```

Comment:
How does that work? When Detach() is called, the container has probably terminated, so the pid does not exist and `/proc/$pid/ns/net` does not exist.

Reply:
Oops. Sorry, I forgot to update this. I will embed the [Tracer](https://github.com/inspektor-gadget/inspektor-gadget/blob/2727379283b37d566b3932bb9f39804329f7640b/pkg/gadgets/internal/networktracer/tracer.go#L99) to reuse the `Detach` and `Close` but reimplementing the `Attach` method. WDYT?

Reply:
Actually, I will not use the `networktracer.Tracer` in this PR as it needs a rework to work also for the network-graph tracer. For this PR, I will just apply the same logic inside the network-graph tracer, but for sure, we could make it work also with the `networktracer.Tracer` in a future PR. WDYT?

Reply:
I agree, let's move forward with this one and then we can generalize `networktracer.Tracer` to make it work on this case too.